rpc/core/src/model/feerate_estimate.rs

```diff
@@ -0,0 +1,51 @@
+use borsh::{BorshDeserialize, BorshSerialize};
+use serde::{Deserialize, Serialize};
+
+#[derive(Clone, Copy, Debug, Serialize, Deserialize, BorshSerialize, BorshDeserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct FeerateBucket {
+    /// The fee/mass ratio estimated to be required for inclusion time <= estimated_seconds
+    pub feerate: f64,
+
+    /// The estimated inclusion time for a transaction with fee/mass = feerate
+    pub estimated_seconds: f64,
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize, BorshSerialize, BorshDeserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct FeeEstimate {
+    /// *Top-priority* feerate bucket. Provides an estimation of the feerate required for sub-second DAG inclusion.
+    pub priority_bucket: FeerateBucket,
+
+    /// A vector of *normal* priority feerate values. The first value of this vector is guaranteed to exist and
+    /// provide an estimation for sub-*minute* DAG inclusion. All other values will have shorter estimation
+    /// times than all `low_bucket` values. Therefor by chaining `[priority] | normal | low` and interpolating
+    /// between them, one can compose a complete feerate function on the client side. The API makes an effort
+    /// to sample enough "interesting" points on the feerate-to-time curve, so that the interpolation is meaningful.
+    pub normal_buckets: Vec<FeerateBucket>,
+
+    /// A vector of *low* priority feerate values. The first value of this vector is guaranteed to
+    /// exist and provide an estimation for sub-*hour* DAG inclusion.
+    pub low_buckets: Vec<FeerateBucket>,
```

Comment:
why not male it depending on request params to only transmit requested data?

Reply:
Wer'e talking about 3-10 data points here at most, so I don't see why this shoud be optimized in any way